src/DynamoCore/Utilities/LuceneSearchUtility.cs

```diff
@@ -269,23 +273,57 @@ internal string CreateSearchQuery(string[] fields, string SearchTerm)
 
             var booleanQuery = new BooleanQuery();
             string searchTerm = QueryParser.Escape(SearchTerm);
+            var bCategoryBasedSearch = searchTerm.Contains('.') ? true : false;
 
             foreach (string f in fields)
             {
+                //Needs to be again due that now a query can contain different values per field (e.g. CategorySplitted:list, Name:tr)
+                searchTerm = QueryParser.Escape(SearchTerm);
+                if (bCategoryBasedSearch == true)
+                {
+                    //This code section should be only executed if the search criteria is CategoryBased like "category.nodename"
+                    if (f != nameof(LuceneConfig.NodeFieldsEnum.NameSplitted) &&
+                        f != nameof(LuceneConfig.NodeFieldsEnum.CategorySplitted))
+                        continue;
+
+                    var categorySearchBased = searchTerm.Split('.');
```

Comment:
@QilongTang for this implementation I'm just considering the base "category.node" search criteria, if we want for support more complex categories we will need think in another solution.
For this case I can add some code to prevent a search of the type "A.B.C" if that's the case we won't return results, what do you think?

Reply:
I've added the validation for supporting the criteria A.B.C, not sure if this is what you expected (if not please put a more detailed example), see the GIF attached with the behavior.
commit: https://github.com/DynamoDS/Dynamo/pull/14663/commits/a3f7bd60ce8964c12caa157c803fa0e7ae800937

![PackagesGuideBugFix2](https://github.com/DynamoDS/Dynamo/assets/61755417/4c4bea49-b420-43ef-8cbf-f68b6a1602d1)